### PR TITLE
Add truncated llama style model init via reset parameters()

### PR DESCRIPTION
**Pull request by lessw2020**
This PR adds the following:
1 - via reset parameters, a full layerwise init for the llama models under /llama.  This uses the total model depth as part of the init via:
self.weight_init_std = 0.02 / (2 * self.num_layers) ** 0.5

2 - The final output ffn (head) is init with sqrt of the dim of the model itself and a slightly wider cutoff factor of 3. 

3 - tangential change - updates run_llama_train.sh with updated MODEL and MODEL_CONF params to allow for direct model control via the sh script.  (there was a MODEL already but it was incorrectly using that in place of MODEL_CONF...though we should update this as it's not intuitive).

4 - made the debugmodel default to 2 layers as an improved debug check.

5 - added a 1B and 40B for additional testing configs.  I can't currently run 70B on my H100 due to OOM, but can run 40B. 

Testing:
Verified proper init and training with 7B, 13B and ~40B:

Screenshot 2024-02-11 at 10 39 12 PM

### CODE REVIEW

File: run_llama_train.sh [hunk 1]
```diff
@@ -22,8 +23,10 @@ CHECKPOINT_FOLDER=${CHECKPOINT_FOLDER:-""}
 # Please adjust this to a longer interval period. The unit of measurement is in steps.
 CHECKPOINT_INTERVAL=${CHECKPOINT_INTERVAL:-5}
 
-torchrun --nproc_per_node=${NGPU} \
+torchrun --nproc_per_node=${NGPU} --rdzv_endpoint="localhost:5972" \
```
Comment (wanchaol) [inline run_llama_train.sh]: we don't need the rdzv endpoint for local runs?

File: torchtrain/train_configs/train_config.toml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 dump_folder = "./torchtrain/outputs"
 
 [profiling]
-run_profiler = true
```
Comment (wanchaol) [inline torchtrain/train_configs/train_config.toml]: any reason turning profiling off?

File: torchtrain/models/llama/model.py [hunk 3]
```diff
@@ -354,9 +381,12 @@ def __init__(self, layer_id: int, args: ModelArgs):
             ffn_dim_multiplier=args.ffn_dim_multiplier,
         )
         self.layer_id = layer_id
+        self.num_layers = args.n_layers
         self.attention_norm = RMSNorm(args.dim, eps=args.norm_eps)
         self.ffn_norm = RMSNorm(args.dim, eps=args.norm_eps)
 
+        self.weight_init_std = 0.02 / (2 * self.num_layers) ** 0.5
```
Comment (wanchaol) [inline torchtrain/models/llama/model.py:395]: I think a good follow up we can do is to use the current layer depth instead of using the global depth

File: run_llama_train.sh [hunk 4]
```diff
@@ -22,8 +23,10 @@ CHECKPOINT_FOLDER=${CHECKPOINT_FOLDER:-""}
...
 --local-ranks-filter ${LOG_RANK} --role rank --tee 3 \
-train.py --steps 10 --compile \
---pp_degree ${PP} --sp_degree ${SP} --dp_degree ${DP}
---checkpoint-folder=${CHECKPOINT_FOLDER} --checkpoint-interval=${CHECKPOINT_INTERVAL}
+train.py --steps 10 \
+--model ${MODEL} --model_conf ${MODEL_CONF} \
+--pp_degree ${PP} --sp_degree ${SP} --dp_degree ${DP} \
+# --compile \
+# --checkpoint-folder=${CHECKPOINT_FOLDER} --checkpoint-interval=${CHECKPOINT_INTERVAL}
```
Comment (wanchaol) [inline run_llama_train.sh]: wondering why turning off checkpoint too?

File: torchtrain/models/llama/model.py [hunk 5]
```diff
@@ -412,6 +450,26 @@ def __init__(self, params: ModelArgs):
 
         self.norm = RMSNorm(params.dim, eps=params.norm_eps)
         self.output = nn.Linear(params.dim, params.vocab_size, bias=False)
+        # init model weights
+        self.reset_parameters()
+        rank0_log(f"{self.params=}")
+
+    def reset_parameters(
```
Comment (wanchaol) [inline torchtrain/models/llama/model.py:465]: the other thing I am wondering is, this PR does not add the actual meta init yet?

File: run_llama_train.sh (hunk 1 above)
Comment (lessw2020) [reply run_llama_train.sh]: yes I'll remove.  I had added it to enable control around stuck ports after OOMs.

File: torchtrain/train_configs/train_config.toml (hunk 2 above)
Comment (lessw2020) [reply torchtrain/train_configs/train_config.toml]: I'll turn it back on.  I had turned it off b/c for this work I'm not looking at the profiles, and the reporting is cluttering up the console while training, and esp for running things like 40B figured it was inefficient to have it on unless I am actually going to use the traces.

File: torchtrain/models/llama/model.py (hunk 3 above)
Comment (lessw2020) [reply torchtrain/models/llama/model.py:395]: oh funny enough I actually did that first, and then figured let's stick with the proven one. 
let's go with this for now as we know it works and trains, and then I can update with current layer depth as part of meta init PR.

File: run_llama_train.sh (hunk 4 above)
Comment (lessw2020) [reply run_llama_train.sh]: same as turning off profiling- it's just using up compute time for no gain while doing the framework build out and esp for 40B don't want to wait for checkpoints that I won't use, to save out. 
I'll turn it back on, but I guess this also calls for adding in the dynamic config we discussed and then we can have a dev config that doesn't have profiling or checkpointing and then a default config that has these things all on for training use.

File: torchtrain/models/llama/model.py (hunk 5 above)
Comment (lessw2020) [reply torchtrain/models/llama/model.py:465]: it's coming as additional PR - I wanted to prove out the core init logic is working and trains from scratch, then will add in the meta init PR next.

File: run_llama_train.sh (hunk 4 above)
Comment (wanchaol) [reply run_llama_train.sh]: Yeah the config files that we take in would be valuable!

File: torchtrain/models/llama/model.py (hunk 3 above)
Comment (wanchaol) [reply torchtrain/models/llama/model.py:395]: haha I think we can put layer-depth or global depth as a init option, the train config would be helpful for user to control such a thing via a init option
